### PR TITLE
Fix scaling issue in DataSet when Grid sampler is used as restart.

### LINKED ISSUE #690: Grid-like datasets are not scaled correctly in the DataSet
--------
Issue Description
--------

When one restart the calculation, the dataset will be scaled before retrieving the realization from that dataset. If grid-like sampler is used, the scale factor could be zero, since the grid points for some variable could be constant, this will cause the failure of the restart capability as recently reported by one of our users. 

##### What did you expect to see happen?
The grid sampler with restart should work.

##### What did you see instead?

From raven user:
```
so I started a test without the meshfile variable. Left it running for some time so I had a dataset of 34 restart points. Then cancled Raven and tried to restart from the previous results but again the same error and Raven ignored the restart points completely and started sampling the same points again that already exist in the restart data.
```

##### Do you have a suggested fix for the development team?
@mandd and I found we have an issue with the SCALE in the DataSet. The fix can be found in 
https://github.com/idaholab/raven/tree/wangc/std_issue 

##### Please attach the input file(s) that generate this error.  The simpler the input, the faster we can find the issue.

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or improvement?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest stable branch? If yes, is there any issue tagged with stable (create if needed)?
- [x] 5. If the issue is being closed without a merge request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by wangcj05**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
close #690 

##### What are the significant changes in functionality due to this change request?
Fix a defect in the DataSet. The Scaling factor could not handle Grid sampling correctly. 

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

### CODE REVIEW

**Comment by wangcj05** [comment]
Why the directory is invalid? @alfoa

**Comment by wangcj05** [comment]
Found the reason 
```In Microsoft Windows operating system, you can't create folders that have their name from one of these words: CON, PRN, AUX, NUL, COM1, COM2, COM3, COM4, COM5, COM6, COM7, COM8, COM9, LPT1, LPT2, LPT3, LPT4, LPT5, LPT6, LPT7, LPT8, and LPT9. ... We creating a folder with the same name would confuse the system.```

**Comment by wangcj05** [comment]
This is ready for you to review @alfoa

File: framework/DataObjects/DataSet.py [hunk 1]
```diff
@@ -1380,6 +1380,10 @@ def _getRealizationFromDataByValue(self,match, tol=1e-15, unpackXArray=False):
         # variable doesn't have a scale factor (yet? Why not?)
           loc = 0.0
           scale = 1.0
+        if scale == 0:
+          # TODO: Seem to me, we need to find a better way to compare data
```
Comment (alfoa) [inline framework/DataObjects/DataSet.py]: @wangcj05 Open an issue...this needs to be discussed

**Comment by alfoa** [comment]
----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code. ***ok***
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts). ***n/a***
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details. ***ok***
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass. ***they pass***
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True. ***test added***
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync. ***n/a***
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done. ***#690***
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added? ***n/a***

**Comment by alfoa** [comment]
Check list passed...

Merging...

**Pull request merged**